lib/openhab/log/logger.rb

```diff
@@ -148,38 +152,26 @@ class << self
       #
       # @return [Logger] for the supplied name
       #
-      def logger(klass)
+      def logger_for_class(klass)
         if klass.respond_to?(:java_class) &&
            klass.java_class &&
            !klass.java_class.name.start_with?('org.jruby.Ruby')
           klass = klass.java_class
         end
-        name = klass.name
+        name = klass.name if klass.name != 'Object'
         @loggers[name] ||= Log.logger_for(name)
       end
 
       #
-      # Configure a logger for the supplied class name
-      #
-      # @param [String] name to configure logger for
-      #
-      # @return [Logger] for the supplied classname
-      #
-      def logger_for(name)
-        configure_logger_for(name)
-      end
-
-      private
-
-      #
-      # Configure a logger for the supplied classname
+      # Configure a logger for the supplied name
       #
       # @param [String] classname to create logger for
       #
-      # @return [Logger] Logger for the supplied classname
+      # @return [Logger] Logger for the supplied name
       #
-      def configure_logger_for(name)
+      def logger_for(name)
```

Comment:
loggers should be constructed in this format:
`jsr223.jruby.<rules_file>.<rule>.<any other classes>`

That lets users set log levels at a rules file level easily. 

The <rule> is optional in case in the logging happens outside of the context of a rule.

Reply:
the entire log name should have spaces replaced with underscores and ruby '::' replaced with '.'  That way people can specify logging levels across the entire logger name.

Reply:
Does this look like what you had in mind:

```
23:28:07.857 [TRACE] [jsr223.jruby.log_test.Module         ] - Defining Java::OrgOpenhabCoreLibraryItems::ContactItem#open? for OPEN
23:28:07.858 [TRACE] [jsr223.jruby.log_test.Module         ] - Defining Java::OrgOpenhabCoreLibraryItems::ContactItem#closed? for CLOSED
23:28:07.995 [TRACE] [HAB.DSL.Between.TimeOfDayRangeElement] - SOD(0) other SOD(86399) Other Class (OpenHAB::DSL::Between::TimeOfDayRangeElement) Result (-1)
23:28:08.026 [TRACE] [jsr223.jruby.log_test.Module         ] - Loaded ACTION: org.openhab.core.model.script.actions.Audio
23:28:08.027 [TRACE] [jsr223.jruby.log_test.Module         ] - Loaded ACTION: org.openhab.core.model.script.actions.Ephemeris
23:28:08.028 [TRACE] [jsr223.jruby.log_test.Module         ] - Loaded ACTION: org.openhab.core.persistence.extensions.PersistenceExtensions
```

```
23:31:19.409 [INFO ] [jsr223.jruby.rule_file.rule_1        ] - A received_command Switch1
```

Reply:
I think what you want is: `jsr223.jruby.rules_file.rule_name.Module`?

What other classes do you have in mind?